pwndbg/gdblib/disasm/arm.py

```diff
@@ -116,10 +158,82 @@ def _memory_string(self, instruction: PwndbgInstruction, op: EnhancedOperand) ->
 
         return f"[{(', '.join(parts))}]"
 
+    def read_thumb_bit(self, instruction: PwndbgInstruction, emu: Emulator) -> int | None:
+        if emu:
+            return emu.read_thumb_bit()
+        elif self.can_reason_about_process_state(instruction):
+            # Read the Thumb bit directly from the process flag register if we can
+            return process_read_thumb_bit()
+        else:
+            return 0
+
     @override
     def _immediate_string(self, instruction, operand):
         return "#" + super()._immediate_string(instruction, operand)
 
+    @override
+    def _read_register(
+        self, instruction: PwndbgInstruction, operand_id: int, emu: Emulator
+    ) -> int | None:
+        # When `pc` is referenced in an operand (typically in a memory operand), the value it takes on
+        # is `pc_at_instruction + 8`. In Thumb mode, you only add 4 to the instruction address.
+        if operand_id == ARM_REG_PC:
+            return instruction.address + (4 if self.read_thumb_bit(instruction, emu) else 8)
+
+        return super()._read_register(instruction, operand_id, emu)
+
+    @override
+    def _parse_memory(
+        self, instruction: PwndbgInstruction, op: EnhancedOperand, emu: Emulator
+    ) -> int | None:
+        """
+        Parse the `ArmOpMem` Capstone object to determine the concrete memory address used.
+
+        Types of memory operands:
+            [Rn]
+            [Rn, #imm]
+            [Rn, Rm]
+            [Rn, Rm, <shift> #imm]
+
+        Capstone represents the object a bit differently then AArch64 to align with the underlying architecture of Arm.
+
+        This representation will change in Capstone 6:
+            https://github.com/capstone-engine/capstone/issues/2281
+            https://github.com/capstone-engine/capstone/pull/1949
+        """
+
+        target = 0
+
+        # All memory operands have `base` defined
+        base = self._read_register(instruction, op.mem.base, emu)
+        if base is None:
+            return None
+
+        if op.mem.base == ARM_REG_PC:
+            # The PC as the base register is a special case - it will align the address to a word (32-bit) boundary
+            # Explanation: https://stackoverflow.com/a/29588678
+            # See "Operation" at the bottom of https://developer.arm.com/documentation/ddi0597/2024-03/Base-Instructions/LDR--literal---Load-Register--literal--
+            base = align_down(4, base)
+
+        target += base
+
+        # Add displacement (zero by default)
+        target += op.mem.disp
```

Comment:
Just
```suggestion
        target = base + op.mem.disp
```

(+ remove previous target assignments)